### DISCUSSION TITLE
Export assignment cannot be used when targeting ECMAScript modules. Consider using 'export default' or another module format instead.

### DISCUSSION THREAD

**Discussion by leanjunio** [pkosiec/mongo-seeding#169] [Q&A] [answered]
I was trying to follow the syntax in the https://github.com/pkosiec/mongo-seeding/blob/master/examples/import-data-ts/example/data/1-categories/testCategories.ts page.

```ts
import faker from "faker";

type User = {
  firstName: string;
  lastName: string;
  email: string;
  mobileNumber: string;
  dob: Date;
  weight: number;
  profession: string;
  age: number;
};

let users: User[];

for (let i = 0; i < 20; i++) {
  const user: User = {
    firstName: faker.name.firstName(),
    lastName: faker.name.lastName(),
    email: faker.internet.email(),
    mobileNumber: faker.phone.phoneNumber(),
    dob: faker.date.past(),
    weight: faker.random.number(),
    profession: "Software Engineer",
    age: faker.random.number(),
  };

  users.push(user);
}

export = users;
```

However, I'm getting the following error:
```
error TS1203: Export assignment cannot be used when targeting ECMAScript modules. Consider using 'export default' or another module format instead.
```

I was going to use `export default ...` but I did see the previous issue.

Is there another way to deal with this?

**Comment by pkosiec** [comment, marked as answer]
Hi @leanjunio,
If you'd like to use this syntax, please make sure that the [`compilerOptions.module` in `tsconfig.json` file](https://www.typescriptlang.org/tsconfig/#module) is set to `commonjs` - like this: https://github.com/pkosiec/mongo-seeding/blob/master/core/tsconfig.json#L5

**Comment by leanjunio** [comment]
Thanks! That helped me out! 

I'm not sure if I should make another ticket for this but, is it possible to relate ObjectIds with this package?

**Reply by pkosiec** [reply]
@leanjunio what do you mean by that? Do you mean make relations between objects? If so, then sure. You can do it using helper `getObjectId` like this: https://github.com/pkosiec/mongo-seeding/blob/master/examples/import-data-ts/example/data/2-posts/posts.ts#L7
or just use ObjectId directly (but you need to make sure it's fixed/predictable in order to make proper relation).

**Reply by leanjunio** [reply]
Oh I see. That's perfect. Thank you
